x/evm/keeper/statedb.go

```diff
@@ -113,28 +113,28 @@ func (k *Keeper) ForEachStorage(ctx sdk.Context, addr common.Address, cb func(ke
 func (k *Keeper) SetBalance(ctx sdk.Context, addr common.Address, amount *big.Int) error {
 	cosmosAddr := sdk.AccAddress(addr.Bytes())
 
-	baseDenom := config.GetDenom()
+	baseDenom := config.GetEVMCoinDenom()
 
-	coin := k.bankKeeper.GetBalance(ctx, cosmosAddr, baseDenom)
+	coin := k.bankWrapper.GetBalance(ctx, cosmosAddr, baseDenom)
 	balance := coin.Amount.BigInt()
 	delta := new(big.Int).Sub(amount, balance)
 	switch delta.Sign() {
 	case 1:
 		// mint
 		coins := sdk.NewCoins(sdk.NewCoin(baseDenom, sdkmath.NewIntFromBigInt(delta)))
-		if err := k.bankKeeper.MintCoins(ctx, types.ModuleName, coins); err != nil {
+		if err := k.bankWrapper.MintCoins(ctx, types.ModuleName, coins); err != nil {
 			return err
 		}
-		if err := k.bankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, cosmosAddr, coins); err != nil {
+		if err := k.bankWrapper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, cosmosAddr, coins); err != nil {
 			return err
```

Comment:
Here we should either:
- use the `MintCoinToAccount`. The pro of this is that we only convert the amount once.
OR
- add the corresponding conversion for the `MintCoins` function in the wrapper. If we choose to do this, we should the `MintCoinToAccount`

Would be cleaner to go with the first option IMO

Reply:
you're right!